### PR TITLE
[Internal][Executor] Extract some common constants to promptflow._constants

### PR DESCRIPTION
**Pull request by PeiwenGaoMS**
# Description

Extract some constants used by both executor server and sdk/cli to `promptflow._constants`.

# All Promptflow Contribution checklist:
- [x] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow/promptflow/_utils/utils.py [hunk 1]
```diff
@@ -366,3 +366,9 @@ def copy_file_except(src_dir, dst_dir, exclude_file):
             src_file_path = os.path.join(root, file)
             dst_file_path = os.path.join(current_dst_dir, file)
             shutil.copy2(src_file_path, dst_file_path)
+
+
+def prepare_folder(path: Union[str, Path]) -> Path:
```
Comment (elliotzh) [inline src/promptflow/promptflow/_utils/utils.py:371]: give it some docstring and more precise name as it is a util function now.

**Review by elliotzh — DISMISSED** [review]

File: src/promptflow/promptflow/_constants.py [hunk 2]
```diff
@@ -164,3 +164,11 @@ class MessageFormatType:
 
 
 DEFAULT_OUTPUT_NAME = "output"
+
+OUTPUT_FILE_NAME = "output.jsonl"
```
Comment (wangchao1230) [inline src/promptflow/promptflow/_constants.py:168]: shall these be kept in promptlfow/_sdk/_constants.py? 

As this storage logic belongs to devkit?

File: src/promptflow/promptflow/_utils/utils.py (hunk 1 above)
Comment (PeiwenGaoMS) [reply src/promptflow/promptflow/_utils/utils.py:371]: Updated

File: src/promptflow/promptflow/_constants.py (hunk 2 above)
Comment (PeiwenGaoMS) [reply src/promptflow/promptflow/_constants.py]: Move all these constants to `_devkit_constants.py`.

**Review by elliotzh — APPROVED** [review]

**Review by elliotzh — APPROVED** [review]

File: src/promptflow/promptflow/_constants.py (hunk 2 above)
Comment (PeiwenGaoMS) [reply src/promptflow/promptflow/_constants.py:168]: This constants are used by executor server,  which belongs to core.

**Review by thy09 — APPROVED** [review]

**Pull request merged**